code/modules/surgery/organs/organ_internal.dm

```diff
@@ -60,6 +60,17 @@
 
 //Special is for instant replacement like autosurgeons
 /obj/item/organ/proc/Remove(mob/living/carbon/M, special = FALSE)
+	//Stop any reagent metabolizing on organ destruction
+	if(reagents && iscarbon(owner))
+		var/mob/living/carbon/body = owner
+		for(var/chem in reagents.reagent_list)
+			var/datum/reagent/reagent = chem
+			if(!reagent.metabolizing)
```

Comment:
the on_mob_delete() call needs to be moved to be BEFORE this check

Reply:
Shit thought i was clever with that early continue.